toolchain/c-toolchain.lisp

```diff
@@ -223,12 +218,23 @@
           *linkkit-start* nil
           *linkkit-end* nil)))
 
+(defun common-toolchain-parameters ()
+  (setf *cc-flags*
+        (append
+         *cc-flags*
+         ;; For MacPorts
+         #+darwin (list "-I" "/opt/local/include/")
+         (split-cflags (getenv "CFLAGS"))
+         ;; FreeBSD non-base header files
+         #+(or freebsd openbsd) (list "-I" "/usr/local/include/"))))
+
 (defun ensure-toolchain-parameters ()
   #+clisp (unless *cc* (clisp-toolchain-parameters))
   #+ecl (unless *cc* (ecl-toolchain-parameters))
   #+mkcl (unless *cc* (mkcl-toolchain-parameters))
   #+sbcl (unless *cc* (sbcl-toolchain-parameters))
-  (unless *cc* (default-toolchain-parameters)))
+  (unless *cc* (default-toolchain-parameters))
+  (common-toolchain-parameters))
```

Comment:
What's going on here? In which case is `*cc*` null? i.e., why do we need `common-toolchain-parameters`?

Reply:
We need `common-toolchain-parameters` because apart from toolchain-specific parameters (such as SBCL-specific parameters based on its `sbcl.mk`) we want to have some common and OS-specific parameters: (1) `*cc-flags*` based on the `CFLAGS` environment variable, (2) `/opt/local/include` on Darwin.

The case of null `*cc*` was handled by `default-toolchain-parameters` before this PR. This PR augments `*cc-flags*` after `*cc-flags*` and `*cc*` were set.

Reply:
Is it that *cc* is being checked to see if ensure-toolchain-parameters has run before?

If so, is it problematic that the new function you've added doesn't check for it?

Reply:
Indeed, there is a surprising second call of ensure-toolchain-parameters here: https://github.com/cffi/cffi/blob/v0.20.1/toolchain/static-link.lisp#L6 . I have moved the body into `unless *cc*`, https://github.com/cffi/cffi/pull/146/commits/5c1f1f67 . (Alternatively the call in `static-link.lisp` may be deleted and `unless *cc*` removed.)